unit-tests/func/serializable-device/test-l500-json-load.py

```diff
@@ -50,48 +86,121 @@
     "Reset Camera Accuracy Health": 0.0,
     "Sensor Mode": 0.0,
     "Trigger Camera Accuracy Health": 0.0,
-    "Visual Preset": 1,
-    "Zero Order Enabled": 0.0,
-    "stream-depth-format": "Z16",
-    "stream-fps": "30",
-    "stream-height": "480",
-    "stream-ir-format": "Y8",
-    "stream-width": "640"
+    "Visual Preset": 1
 }
 """
 
-def log_settings_differences():
-    global data, depth_sensor, sd
-    depth_sensor.set_option(rs.option.visual_preset, int(rs.l500_visual_preset.low_ambient_light))
-    actual_data = str( sd.serialize_json() )
-    log.debug_indent()
-    try:
-        # logging the differences in the settings between the expected and the actual values
-        log.d( "Printing differences between expected and actual settings values:" )
-        for expected_line, actual_line in zip( data.splitlines()[2:-1], actual_data.splitlines()[1:-1] ):
-            # data starts with an empty line and a '{' and ends with a '}', so we cut out the first 2 lines and the last
-            # line to ignore them. Same with the actual data ony it doesn't start with an empty line
-            if "Visual Preset" in expected_line or "Temperature" in expected_line or "temperature" in expected_line:
-                # the line regarding the visual preset will always be different because we load 1 from data but set it to
-                # 3 for low ambient. Also all lines regarding temperatures depend on the camera and don't affect the preset
-                continue
-            if expected_line != actual_line:
-                log.d( "Expected to find line:", expected_line)
-                log.d( "But found:            ", actual_line)
-    finally:
-        log.debug_unindent()
-
 test.start("Trying to load default settings from json")
 try:
-    sd.load_json(data)
+    sd.load_json( low_ambient_data_with_default_preset )
     visual_preset_number = depth_sensor.get_option(rs.option.visual_preset)
     visual_preset_name = rs.l500_visual_preset(int(visual_preset_number))
 
     # if this check fails it is most likely because FW changed the default settings
     equal = test.check_equal(visual_preset_name, rs.l500_visual_preset.low_ambient_light)
     if not equal:
         log.w( "It is possible that FW changed the default settings of the camera." )
-        log_settings_differences()
+        log_settings_differences( low_ambient_data_with_default_preset )
+except:
+    test.unexpected_exception()
+test.finish()
+
+#############################################################################################
+# this test checks that if the given settings are wrong (don't fit any preset), and we use default preset
```

Comment:
I think a better explanation is in order:
```
# There is no default preset: so when one is specified, the code should calculate the preset!
# Here, we intentionally should not fit into any of the defined presets, and check that the result is CUSTOM
```

Reply:
done